Content.Client/SubFloor/TrayScannerSystem.cs

```diff
@@ -130,36 +147,58 @@ private bool UpdateTrayScanner(EntityUid uid, TrayScannerComponent? scanner = nu
         // is still technically on
         if (flooredPos == Vector2.Zero)
         {
-            _subfloorSystem.ToggleSubfloorEntities(scanner.RevealedSubfloors, false, uid, _visualizerKeys);
+            _subfloorSystem.SetEntitiesRevealed(scanner.RevealedSubfloors, uid, false, _visualizerKeys);
             scanner.RevealedSubfloors.Clear();
             return true;
         }
 
+        // TODO REDO THIS
+        // Here we avoid the entity lookup & return early if the scanner's position hasn't appreciably changed. However,
+        // if a new player enters PVS-range, they will update the in-range entities on their end and use that to set
+        // LastLocation. This means that different players can technically see different entities being revealed by the
+        // same scanner. The correct fix for this is probably just to network the revealed entity set.... But I CBF
+        // doing that right now....
+        //
+        // Though this approach also has other issues. If the scanner is on a shuttle, and another grid moves close it,
+        // the scanner's local position doesn't update. But if you based this on the scanner's world position it would
+        // unnecessary update whenever the grid that it is on is moving. I guess you could just restrict scanners to
+        // only scan one grid at a time?
```

Comment:
At the time we talked about it and agree it's not worth trying to accommodate. I think ideally we probably shouldn't ever show stuff on another grid.

Reply:
Yeah it probably isn't worth the effort. I'll just make scanners only work on-grid. Might be a bit annoying for engineers working on the outside edge... but they can just step on a tile.